test/test_crypten.py

```diff
@@ -57,16 +57,19 @@ def test_cat_stack(self):
         encrypted1 = crypten.cryptensor(tensor1)
         encrypted2 = crypten.cryptensor(tensor2)
 
-        for op in ["cat", "stack"]:
-            reference = getattr(torch, op)([tensor1, tensor2])
-            encrypted_out = getattr(crypten, op)([encrypted1, encrypted2])
-            self._check(encrypted_out, reference, "%s failed" % op)
-
-            for dim in range(4):
-                reference = getattr(torch, op)([tensor1, tensor2], dim=dim)
-                encrypted_out = getattr(crypten, op)([encrypted1, encrypted2], dim=dim)
+        for module in [crypten, torch]:  # torch.cat on CrypTensor runs crypten.cat
+            for op in ["cat", "stack"]:
+                reference = getattr(torch, op)([tensor1, tensor2])
```

Comment:
Q: Shouldn't this be called on ```encrpyted1``` and ```encrypted2```? ```torch.cat(encrypted_tensors) --> crypten.cat(encryptend_tensors)```

Reply:
I don't think so? Line 62 is running the PyTorch reference implementation on PyTorch tensors. Line 63 is running the corresponding CrypTen implementation on `CrypTensor`s (testing both the `crypten.{cat,stack}` and `torch.{cat,stack}` code paths for `CrypTensor`). Line 64 checks that, after decryption, the CrypTen result matches the PyTorch result.

Reply:
Yep, you are right!
I do not know why I did not see the ```for module in [crypten, torch]```.